Hide Link header on certain screens

tillh-stripe opened the pull request:
# Summary
<!-- Simple summary of what was changed. -->

This pull request hides the Link header in ~screens that are not the root screen~ certain screens, as shown in Figma.

We achieve this by animating the alpha of the header content, which results in a smooth transition between screens.

# Motivation
<!-- Why are you making this change? If it's for fixing a bug, if possible, please include a code snippet or example project that demonstrates the issue. -->

Alignment with iOS.

# Testing
<!-- How was the code tested? Be as specific as possible. -->
- [x] Added tests
- [ ] Modified tests
- [x] Manually verified

# Screenshots
| Before  | After |
| ------------- | ------------- |
| ![Screenshot_20220822_114002](https://user-images.githubusercontent.com/110940675/185961909-0eb0bca5-072b-4851-a3a9-5908157e17e0.png) | ![Screenshot_20220822_114307](https://user-images.githubusercontent.com/110940675/185963887-a50557fb-025f-4b5d-91b8-1470c023087d.png) |

# Screen recording

https://user-images.githubusercontent.com/110940675/186027661-9433d2d0-91e9-45f1-849c-e201d7ed8e6a.mp4

# Changelog
<!-- Is this a notable change that affects users? If so, add a line to `CHANGELOG.md` and prefix the line with one of the following:
    - [Added] for new features.
    - [Changed] for changes in existing functionality.
    - [Deprecated] for soon-to-be removed features.
    - [Removed] for now removed features.
    - [Fixed] for any bug fixes.
    - [Security] in case of vulnerabilities.
-->

tillh-stripe commented on link/src/main/java/com/stripe/android/link/ui/Common.kt:
```diff
@@ -19,16 +21,30 @@ typealias BottomSheetContent = @Composable ColumnScope.() -> Unit
+    omitTopPadding: Boolean = false,
     content: @Composable ColumnScope.() -> Unit
 ) {
+    val padding = remember(omitTopPadding) {
```
Added this, because the spacing at the top of the screen looks fairly large now in the sub-screens.

brnunes-stripe commented on link/src/main/java/com/stripe/android/link/ui/LinkAppBar.kt:
```diff
@@ -107,3 +100,59 @@ internal fun LinkAppBar(
+
+@Preview
+@Composable
+internal fun LinkAppBar() {
```
Can these be private instead of internal?

tillh-stripe replied:
Yep, changed!

brnunes-stripe reviewed (changes requested):
I think this will hide the top bar in a few situations where we don't want to, like when you're signing up and navigate to the verification screen.
Can you verify it matches all cases from Figma?

tillh-stripe commented:
@brnunes-stripe Yep, you’re right. I misunderstood the Jira issue. Updated the logic now to only hide it in the card edit and add payment method screens.

I myself was getting confused by the logic, so to make it easier to understand, I created a `LinkAppBarState` and added tests to make sure it’s constructed correctly. Let me know what you think!
